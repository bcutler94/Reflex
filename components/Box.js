import React from "react";
import { StyleSheet, TouchableOpacity } from 'react-native';


export const Box = ({ shouldFlash = false, handlePress }) => {

    const styleToUse = shouldFlash ? { ...styles.box, backgroundColor: "blue" } : styles.box;
    return (
        <TouchableOpacity style={styleToUse} onPress={handlePress} />
    )
}

const styles = StyleSheet.create({
    box: {
        alignItems: 'center',
        justifyContent: 'center',
        aspectRatio: 1,
        height: '24%',
        marginBottom: "1%",
        backgroundColor: "lightgray"
    },
});