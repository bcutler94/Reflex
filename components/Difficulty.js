import React, { useState } from "react";
import { StyleSheet, Text, TouchableOpacity, View} from 'react-native';

const LEVELS = ['easy', "medium", "hard"];

export const Difficulty = ({ handleParentPress }) => {

    const [diff, setDiff] = useState('easy');

    const handlePress = (diff) => {
        setDiff(diff)
        handleParentPress(diff)
    }

    return (
        <View style={styles.container}>
            {
                LEVELS.map(level => {
                    const touchableStyleToUse = diff === level ? {...styles.button, borderColor: "green", backgroundColor: 'yellow'} : styles.button;
                    const textStyleToUse = diff === level ? {...styles.text, fontSize: 25, color: "green"} : styles.text;
                    return (
                        <TouchableOpacity style={touchableStyleToUse} onPress={() => handlePress(level)}>
                            <Text style={textStyleToUse}>
                                {
                                    level.toUpperCase()
                                }
                            </Text>
                        </TouchableOpacity>
                    )
                })
            }
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
        flexDirection: 'row',
        width: "100%",
        justifyContent: 'space-evenly'
    },
    button: {
        alignItems: 'center',
        justifyContent: 'center',
        aspectRatio: 2.5,
        height: 50,
        borderRadius: 10,
        borderWidth: 2
    },
    text: {
        fontSize: 20
    }
});