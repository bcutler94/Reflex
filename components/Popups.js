import React from "react";
import { StyleSheet, Text, TouchableOpacity, View, Modal, Dimensions } from 'react-native';


export const GameOver = ({ visible = false, timeElapsed, handleModalDismiss, highScore }) => {
    return (
        <Modal
        transparent
        presentationStyle="overFullScreen"
        animationType="slide"
        visible={visible}
        >
        <View style={styles.modalContainer}>
            <View style={styles.modal}>
                <Text>
                    Game Over!
                </Text>
                <Text>
                    {
                        `Your high score was ${highScore}`
                    }
                </Text>
                <Text>
                    {
                        `Your time was ${timeElapsed}`
                    }
                </Text>
                <TouchableOpacity onPress={handleModalDismiss} style={styles.button}>
                    <Text>
                        RESET GAME
                    </Text>
                </TouchableOpacity>
            </View>
    
        </View>
        </Modal>
    )
}

export const FTUE = ({ visible, handleFtueDismiss }) => {
    return (
        <Modal
        transparent
        presentationStyle="overFullScreen"
        animationType="slide"
        visible={visible}
        >
        <View style={styles.modalContainer}>
            <View style={styles.modal}>
                <Text style={styles.title}>
                    Welcome to REFLEX!
                </Text>
                <Text style={styles.body}>
                    The objective of this game is to test your reflex speed. 
                    Once you start the game, you must press the squares that flash a different color.
                    The speed at which squares flash, as well as the reaction time alloted will get more difficult with time.
                    Good Luck!
                </Text>
                <TouchableOpacity onPress={handleFtueDismiss} style={styles.button}>
                    <Text>
                        OK
                    </Text>
                </TouchableOpacity>
            </View>
    
        </View>
        </Modal>
    )
}

const { width } = Dimensions.get("screen");

const styles = StyleSheet.create({
    modalContainer: {
        flex: 1,
        alignItems: 'center',
        justifyContent: 'center'
    },
    modal: {
        alignItems: 'center',
        justifyContent: 'space-evenly',
        aspectRatio: 1,
        width: "80%",
        borderRadius: 10,
        borderWidth: 3,
        backgroundColor: 'white',
        padding: 10
    },
    button: {
        backgroundColor: 'green',
        aspectRatio: 5,
        width: "80%",
        justifyContent: 'center',
        alignItems: 'center',
        borderRadius: 10
    },
    title: {
        fontSize: width * .05,
        fontWeight: '800',
        textAlign: 'center'
    },
    body: {
        fontSize: width * .04,
        textAlign: 'center'
    }
  });