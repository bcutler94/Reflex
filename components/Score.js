import React from "react";
import { StyleSheet, Text, View} from 'react-native';


export const Score = ({ score = null }) => {

    return (
        <View style={styles.container}>
            <Text>
                {
                    `SCORE: ${score}`
                }
            </Text>
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
      backgroundColor: 'lightblue',
      justifyContent: 'center',
      alignItems: 'center',
      borderRadius: 10,
      aspectRatio: 3,
      width: '30%'
    },
  });