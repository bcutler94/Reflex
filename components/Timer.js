import React, { useEffect, useState } from "react";
import { StyleSheet, Text, View } from 'react-native';
import moment from 'moment';

const getDuration = (startTime) => new moment (new moment ().diff(startTime)).format("m:ss:SS")
const startTime = new moment ();

export const Timer = ({ startTimer, handleParentTime }) => {

    const [time, setTime] = useState(getDuration(new moment ()))

    useEffect(() => {
        let timer;
        if (startTimer) {
            const startTime = new moment ();
            timer = setInterval(() => setTime(getDuration(startTime)), 10);
        } else {
            handleParentTime?.(time)
            timer && clearTimeout(timer);
        }
        return () => clearTimeout(timer);
    }, [startTimer])


    return (
        <View style={styles.container}>
            <Text>
                {
                    `TIME: ${time}`
                }
            </Text>
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
      backgroundColor: 'lightblue',
      justifyContent: 'center',
      alignItems: 'center',
      borderRadius: 10,
      aspectRatio: 3,
      width: '30%'
    },
});