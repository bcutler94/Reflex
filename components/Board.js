import React from "react";
import { StyleSheet, View } from 'react-native';
import { Box } from "./Box";


export const Board = ({ flashBoxes, flash, handlePress }) => {
    return (
        <View style={styles.board}>
            {
                new Array(16).fill(null).map((_, i) => {
                    const shouldFlash = flashBoxes && flashBoxes.has(i) && flash;
                    return (
                        <Box 
                            key={i} 
                            handlePress={() => handlePress(i)}
                            shouldFlash={shouldFlash}
                        /> 
                    )
                })
            }
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#fff',
        alignItems: 'center',
        justifyContent: 'space-evenly',
      },
    board: {
        aspectRatio: 1,
        width: "95%",
        flexWrap: "wrap",
        flexDirection: 'row',
        justifyContent: 'space-between',
        alignItems: 'center'
    },
  });