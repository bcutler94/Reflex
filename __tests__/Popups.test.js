import React from 'react';
import renderer from 'react-test-renderer';
import { FTUE, GameOver } from '../components/Popups';

describe('<Poups />', () => {
    it('FTUE renders correctly', () => {
        const tree = renderer.create(<FTUE />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('GameOver renders correctly', () => {
        const tree = renderer.create(<GameOver />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});