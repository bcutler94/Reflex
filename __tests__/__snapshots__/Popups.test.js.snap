// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Poups /> FTUE renders correctly 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  presentationStyle="overFullScreen"
  transparent={true}
  visible={true}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "aspectRatio": 1,
          "backgroundColor": "white",
          "borderRadius": 10,
          "borderWidth": 3,
          "justifyContent": "space-evenly",
          "padding": 10,
          "width": "80%",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 37.5,
            "fontWeight": "800",
            "textAlign": "center",
          }
        }
      >
        Welcome to REFLEX!
      </Text>
      <Text
        style={
          Object {
            "fontSize": 30,
            "textAlign": "center",
          }
        }
      >
        The objective of this game is to test your reflex speed. Once you start the game, you must press the squares that flash a different color. The speed at which squares flash, as well as the reaction time alloted will get more difficult with time. Good Luck!
      </Text>
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "aspectRatio": 5,
            "backgroundColor": "green",
            "borderRadius": 10,
            "justifyContent": "center",
            "opacity": 1,
            "width": "80%",
          }
        }
      >
        <Text>
          OK
        </Text>
      </View>
    </View>
  </View>
</Modal>
`;

exports[`<Poups /> GameOver renders correctly 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  presentationStyle="overFullScreen"
  transparent={true}
  visible={false}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "aspectRatio": 1,
          "backgroundColor": "white",
          "borderRadius": 10,
          "borderWidth": 3,
          "justifyContent": "space-evenly",
          "padding": 10,
          "width": "80%",
        }
      }
    >
      <Text>
        Game Over!
      </Text>
      <Text>
        Your high score was undefined
      </Text>
      <Text>
        Your time was undefined
      </Text>
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "aspectRatio": 5,
            "backgroundColor": "green",
            "borderRadius": 10,
            "justifyContent": "center",
            "opacity": 1,
            "width": "80%",
          }
        }
      >
        <Text>
          RESET GAME
        </Text>
      </View>
    </View>
  </View>
</Modal>
`;
